api/v1alpha1/etcdbackup_types.go

```diff
@@ -1,25 +1,92 @@
 package v1alpha1
 
 import (
+	corev1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
 )
 
 // EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
 // NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
 
+type EtcdBackupPhase string
+
+var (
+	EtcdBackupPhaseBackingUp EtcdBackupPhase = "BackingUp"
+	EtcdBackupPhaseUploading EtcdBackupPhase = "Uploading"
+	EtcdBackupPhaseCompleted EtcdBackupPhase = "Completed"
+	EtcdBackupPhaseFailed    EtcdBackupPhase = "Failed"
+)
+
+// EtcdClusterEndpoint holds an addressable endpoint for an etcd member.
+type EtcdClusterEndpoint struct {
+	// Port that is exposing the etcd client API for this member.
+	Port intstr.IntOrString `json:"port"`
+	// An IP address or DNS name of an endpoint.
+	Host string `json:"host"`
+	// Scheme to use for connecting to the host.
+	Scheme corev1.URIScheme `json:"scheme"`
+}
+
+// EtcdBackupDestination holds a storage location where an etcd backup will be placed.
+// At most one destination must be set.
+type EtcdBackupDestination struct {
+	// Local, when set, will copy the backup into a local volume on the pod taking the backup.
+	// +optional
+	Local *EtcdBackupDestinationLocal `json:"local,omitempty"`
+	// GCSBucket, when set, will push the backup to a Google Cloud Storage bucket.
+	// +optional
+	GCSBucket *EtcdBackupDestinationGCSBucket `json:"gcsBucket,omitempty"`
+}
+
+// EtcdBackupDestinationLocalVolume describes a volume on the local machine where the backup will be placed.
```

Comment:
Clarify that this just writes into the Pod's filesystem. "Machine" could mislead someone to thinking that we're automatically finding a way to write to the node.

Also correct name of struct.

```suggestion
// EtcdBackupDestinationLocal describes a local directory into which to put the backup file. This is in the filesystem of the pod running the operator. To persist this between pod restarts, ensure that path is inside a mounted volume.
```